README.md

```diff
@@ -1,20 +1,88 @@
 # Flow Data Provisioning Service
 
+The Flow Data Provisioning Service aims at providing a scalable and efficient way to access the execution state of past
+and live sporks, as well as implementing the Rosetta API standard.
+
+Past sporks are loaded from the filesystem using Flow's Write-Ahead Log to replay the changes to all registries, and
+updates are pushed directly to the DPS by the execution nodes in order to handle live networks.
+
+The Flow DPS maintains different specialized indexes for different purposes. One index is used for accessing the entire
+execution state at any given height, while the other is used to follow the history of a specific registry through time.
+Contrary to Flow's WAL, those indexes allow replaying changes both backward and forward, which enables visualization
+beyond a network's pruned blocks.
+
+Another role of the DPS is to provide information about atomic transactions which are not normally generated by Flow, by
+locally running Cadence scripts on the copy of the execution state. This makes it possible to link an account with its
+staked balance, locked token vaults etc. and to expose that information in the API.
```

Comment:
`The DPS will also support a set of custom proxy resources for token vaults and staking/delegating resources. This will allow the DPS to track account balances for locked, staked and delegated tokens for accounts which deploy these custom proxy resources.`